junit5-engine/src/main/java/org/junit/gen5/engine/junit5/descriptor/JUnit5TestableFactory.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.junit.gen5.engine.junit5.descriptor;
+
+import static org.junit.gen5.commons.util.ReflectionUtils.loadClass;
+
+import java.lang.reflect.AnnotatedElement;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.junit.gen5.commons.util.Preconditions;
+import org.junit.gen5.commons.util.ReflectionUtils;
+
+public class JUnit5TestableFactory {
+
+	private static final String SEPARATORS = ":$#";
+
+	public JUnit5Testable fromUniqueId(String uniqueId, String engineId) {
+		Preconditions.notEmpty(uniqueId, "uniqueId must not be empty");
+		List<String> parts = split(uniqueId);
+		Preconditions.condition(parts.remove(0).equals(engineId), "uniqueId must start with engineId");
```

Comment:
> One example: Let's say we have two engines that have a conflicting local/engine-specific unique id (other cases dont lead to problems anyways). What could a Launcher do to make them unique? The only feasible solution is to add the engine id in front of the local/engine-specific id. If there is still a conflict, the Launcher must give up and throw an exception telling that the engines are not compatible. Why?
> 

If the launcher prefixes a "locally unique ID" with a _unique_ engine ID, how could the resulting ID be NOT globally unique?